src/roles/succubus.py

```diff
@@ -0,0 +1,315 @@
+                pm(cli, e, messages["entranced_revert_win"])
+        elif ENTRANCED:
+            msg = []
+            # Run in two loops so we can play the message for everyone dying at once before we actually
+            # kill any of them off (if we killed off first, the message order would be wrong wrt death chains)
+            comma = ""
+            if var.ROLE_REVEAL in ("on", "team"):
+                comma = ","
+            for e in ENTRANCED:
+                if var.ROLE_REVEAL in ("on", "team"):
+                    role = get_reveal_role(e)
+                    an = "n" if role.startswith(("a", "e", "i", "o", "u")) else ""
+                    msg.append("\u0002{0}\u0002, a{1} \u0002{2}\u0002".format(e, an, role))
+                else:
+                    msg.append("\u0002{0}\u0002".format(e))
+            if len(msg) == 1:
+                cli.msg(botconfig.CHANNEL, messages["succubus_die_kill"].format(msg[0] + comma))
+            elif len(msg) == 2:
+                cli.msg(botconfig.CHANNEL, messages["succubus_die_kill"].format(msg[0] + comma + " and " + msg[1] + comma))
+            else:
+                cli.msg(botconfig.CHANNEL, messages["succubus_die_kill"].format(", ".join(msg[:-1]) + ", and " + msg[-1] + comma))
+            for e in ENTRANCED:
+                # to ensure we do not double-kill someone, notify all child deaths that we'll be
+                # killing off everyone else that is entranced so they don't need to bother
+                dlc = list(evt.params.deadlist)
+                dlc.extend(ENTRANCED - {e})
+                debuglog("{0} ({1}) SUCCUBUS DEATH KILL: {2} ({3})".format(nick, nickrole, e, get_role(e)))
+                evt.params.del_player(cli, e, end_game=False, killer_role="succubus",
+                    deadlist=dlc, original=evt.params.original, ismain=False)
+                evt.data["pl"] = evt.params.refresh_pl(evt.data["pl"])
+        ENTRANCED_DYING.clear()
+
+@event_listener("transition_day_resolve", priority=1)
+def on_transition_day_resolve(evt, cli, var, victim):
+    if victim in var.ROLES["succubus"] and VISITED.get(victim) and victim not in evt.data["dead"] and victim in evt.data["onlybywolves"]:
+        # TODO: check if this is necessary for succubus, it's to prevent a message playing if alpha bites
+        # a harlot that is visiting a wolf, since the bite succeeds in that case.
+        if victim not in evt.data["bitten"]:
+            evt.data["message"].append(messages["target_not_home"])
+            evt.data["novictmsg"] = False
+        evt.stop_processing = True
+        evt.prevent_default = True
+
+@event_listener("transition_day_resolve_end", priority=1)
+def on_transition_day_resolve_end(evt, cli, var, victims):
+    for victim in victims + evt.data["bitten"]:
+        if victim in evt.data["dead"] and victim in VISITED.values() and (victim in evt.data["bywolves"] or victim in evt.data["bitten"]):
+            for succ in VISITED.keys():
```

Comment:
No need to use `.keys()` here, just iterate over the dict. Or use `.items()` directly, thus meaning you don't need to access the dict again for the value.